src/parser/types.rs

```diff
@@ -401,12 +392,25 @@ impl FnSig {
 }
 
 /// Any kind of text that can represent a type, a variable, a function name, etc.
-#[derive(Debug, Default, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[derive(Debug, Default, Clone, Eq, Serialize, Deserialize)]
 pub struct Ident {
     pub value: String,
     pub span: Span,
 }
 
+impl PartialEq for Ident {
+    fn eq(&self, other: &Self) -> bool {
+        self.value == other.value
+    }
+}
+
+impl Hash for Ident {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.value.hash(state);
+        self.span.hash(state);
```

Comment:
I think the TODO was explicitly about not having the span in the hash :o

Reply:
btw I think this solution would be cleaner: https://users.rust-lang.org/t/rust-crate-for-ignore-this-field-for-eq-partialeq-hash/33651/6

Reply:
Hmm I see, I misunderstood the intention. I can fix it in this PR (or restore the TODO if that's preferable).

I think the `derivative` crate is [unmaintained](https://github.com/mcarton/rust-derivative/issues/117) and still uses syn v1, so that may not be a great idea. However an alternative ([educe](https://github.com/magiclen/educe)) was also presented in that issue, and that uses syn v2. We can bring that in, if this is preferable over manual implementation(s).

Reply:
Using `educe` in [dfc03ad](https://github.com/zksecurity/noname/pull/108/commits/dfc03ad2a8e00558122e236de4f1ddb44cb652c5)